src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs

```diff
@@ -281,33 +282,41 @@ public void Write(Utf8JsonWriter writer)
                 writer.WritePropertyName("tags");
                 writer.WriteStartObject();
 
+                var originalUICulture = Thread.CurrentThread.CurrentUICulture;
+                Thread.CurrentThread.CurrentUICulture = CultureInfo.InvariantCulture;
+
                 foreach (var tag in this.Tags.Value)
                 {
-                    if (tag.Value is int intValue)
-                    {
-                        writer.WriteString(tag.Key, intValue.ToString(CultureInfo.InvariantCulture));
-                    }
-                    else if (tag.Value is double doubleVal)
-                    {
-                        writer.WriteString(tag.Key, doubleVal.ToString(CultureInfo.InvariantCulture));
-                    }
-                    else if (tag.Value is string stringVal)
+                    switch (tag.Value)
                     {
-                        writer.WriteString(tag.Key, stringVal);
-                    }
-                    else
-                    {
-                        // Should we try to convert to string? Or
-                        // just drop it?
-                        writer.WriteString(tag.Key, tag.Value.ToString());
+                        case string stringVal:
+                            writer.WriteString(tag.Key, stringVal);
+                            break;
+                        case int[] intArrayValue:
+                            writer.WriteString(tag.Key, string.Join(",", intArrayValue));
+                            break;
+                        case double[] doubleArrayValue:
+                            writer.WriteString(tag.Key, string.Join(",", doubleArrayValue));
+                            break;
+                        case bool[] boolArrayValue:
+                            writer.WriteString(tag.Key, string.Join(",", boolArrayValue));
+                            break;
+                        default:
+                            // Should we try to convert to string? Or
```

Comment:
It was there before but I think we can remove this comment It is a best-effort and we don't want to crash here anyway.

Reply:
removed! :)